lib/klaviyo/client.rb

```diff
@@ -40,10 +40,19 @@ def self.request(method, path, content_type, **kwargs)
     end
 
     def self.public_request(method, path, **kwargs)
-      check_public_api_key_exists()
-      params = build_params(kwargs)
-      url = "#{BASE_API_URL}/#{path}?#{params}"
-      res = Faraday.get(url).body
+      check_public_api_key_is_valid()
+      if method == HTTP_GET
+        params = build_params(kwargs)
+        url = "#{BASE_API_URL}/#{path}?#{params}"
+        res = Faraday.get(url).body
+      elsif method == HTTP_POST
```

Comment:
would it make sense to add an `else` here to throw an error if someone specifies an invalid method?

Reply:
Good idea I'll add that 👍